doc/source/fq_nmod_sparse_vec.rst

```diff
@@ -0,0 +1,176 @@
+-------------------------------------------------------------------------------
+
+.. type:: fq_nmod_sparse_entry_t
+
+    Holds a pair (ind, val), where ind (of type slong) is an index into the
+    vector and val (of type fq_nmod_t) is the value at that index
+
+.. type:: fq_nmod_sparse_vec_t
+
+    Holds an array of nonzero entries (of type sparse_entry_struct *) of 
+    specified length nnz, sorted by ind
+
+Memory management
+--------------------------------------------------------------------------------
+
+
+.. function:: void fq_nmod_sparse_vec_init(fq_nmod_sparse_vec_t vec, const fq_nmod_ctx_t ctx)
+
+    Initializes an empty sparse vector (no allocation)
+
+.. function:: void fq_nmod_sparse_vec_clear(fq_nmod_sparse_vec_t vec, slong len, const fq_nmod_ctx_t ctx)
+
+    Clears the entries of vec and frees the space allocated for it
+
+.. function:: void fq_nmod_sparse_vec_swap(fq_nmod_sparse_vec_t vec1, fq_nmod_sparse_vec_t vec2, const fq_nmod_ctx_t ctx)
+
+    Swaps two vectors (no reallocaton)
+
+Instantiation
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_vec_zero(fq_nmod_sparse_vec_t vec, const fq_nmod_ctx_t ctx)
+
+    Sets vec to zero (the empty sparse vector)
+
+.. function:: void fq_nmod_sparse_vec_one(fq_nmod_sparse_vec_t vec, slong ind, const fq_nmod_ctx_t ctx)
+
+    Sets vec to the ind-th basis vector (single one in position ind)
+
+.. function:: void fq_nmod_sparse_vec_set(fq_nmod_sparse_vec_t dst, fq_nmod_sparse_vec_t src, const fq_nmod_ctx_t ctx)
+
+    Makes dst a (deep) copy of src
+
+.. function:: void fq_nmod_sparse_vec_set_entry(fq_nmod_sparse_vec_t vec, slong ind, const fq_nmod_t val, const fq_nmod_ctx_t ctx)
+
+    Sets the value at index ind to val, either replacing an existing value or extending
+    the array of entries
+
+.. function:: void _fq_nmod_sparse_vec_from_entries(fq_nmod_sparse_vec_t vec, slong *inds, fq_nmod_struct *vals, slong nnz, const fq_nmod_ctx_t ctx)
+
+    Constructs vec from a given sequence of indices and associated values, both of length nnz.
+    Assumes no duplicate indices
+
+Comparison
+--------------------------------------------------------------------------------
+
+.. function:: void fq_nmod_sparse_is_zero(fq_nmod_sparse_vec_t vec, const fq_nmod_ctx_t ctx)
+
+    Checks if the given vector is trivial (empty), returning `1` if so and `0` 
+    otherwise
+
+.. function:: void fq_nmod_sparse_vec_equal(const fq_nmod_sparse_vec_t vec1, const fq_nmod_sparse_vec_t vec2, slong ioff, const fq_nmod_ctx_t ctx)
+
+    Checks if vec1 equals vec2 (with s specified column offset ioff), returning
+    `1` if so and `0` otherwise
+
+Indexing
+--------------------------------------------------------------------------------
+
+.. function:: fq_nmod_t * fq_nmod_sparse_vec_at(fq_nmod_sparse_vec_t vec, slong ind, const fq_nmod_ctx_t ctx)
+
+    Returns a pointer to the value at the index ind (or NULL if index not found)
```

Comment:
We tend to use the word "ref" rather than "at" for pointers to entries. Perhaps compare with some other examples in Flint for something more expected.